tests/boulder-integration.sh

```diff
@@ -37,14 +37,41 @@ common() {
 # unrequested challenge to prevent regressions in #3601.
 python -m SimpleHTTPServer $http_01_port &
 python_server_pid=$!
-common --domains le1.wtf --preferred-challenges tls-sni-01 auth
+
+export HOOK_TEST="/tmp/hook$$"
+common --domains le1.wtf --preferred-challenges tls-sni-01 auth \
+       --pre-hook 'echo wtf.pre >> "$HOOK_TEST"' \
+       --post-hook 'echo wtf.post >> "$HOOK_TEST"'\
+       --renew-hook 'echo renew >> "$HOOK_TEST"'
 kill $python_server_pid
 python -m SimpleHTTPServer $tls_sni_01_port &
 python_server_pid=$!
-common --domains le2.wtf --preferred-challenges http-01 run
+common --domains le2.wtf --preferred-challenges http-01 run \
+       --pre-hook 'echo wtf.pre >> "$HOOK_TEST"' \
+       --post-hook 'echo wtf.post >> "$HOOK_TEST"'\
+       --renew-hook 'echo renew >> "$HOOK_TEST"'
 kill $python_server_pid
 
-common -a manual -d le.wtf auth --rsa-key-size 4096
+common -a manual -d le.wtf auth --rsa-key-size 4096 \
+       --pre-hook 'echo wtf2.pre >> "$HOOK_TEST"' \
+       --post-hook 'echo wtf2.post >> "$HOOK_TEST"'
+
+CheckHooks() {
+    EXPECTED="/tmp/expected$$"
+    echo "wtf.pre" > "$EXPECTED"
+    echo "wtf2.pre" >> "$EXPECTED"
+    echo "renew" >> "$EXPECTED"
+    echo "renew" >> "$EXPECTED"
+    echo "wtf.post" > "$EXPECTED"
+    echo "wtf2.post" >> "$EXPECTED"
+    if cmp --quiet "$EXPECTED" "$HOOK_TEST" ; then
```

Comment:
I'm a bit worried about this test because according to the [glob documentation](https://docs.python.org/2/library/glob.html) which [we use for finding renewal configuration files](https://github.com/certbot/certbot/blob/master/certbot/storage.py#L29), the order returned is arbitrary meaning whether `wtf.pre` or `wtf2.pre` comes first isn't well defined.